Tests/BackendIntegrationTests/StoreKitIntegrationTests.swift

```diff
@@ -433,10 +453,97 @@ class StoreKit1IntegrationTests: BaseBackendIntegrationTests {
 
 }
 
-private extension StoreKit1IntegrationTests {
+// MARK: - Observer Mode tests
+
+class BaseStoreKitObserverModeIntegrationTests: BaseStoreKitIntegrationTests {
+
+    override class var observerMode: Bool { return true }
+
+}
+
+class StoreKit2ObserverModeIntegrationTests: StoreKit1ObserverModeIntegrationTests {
+
+    override class var storeKit2Setting: StoreKit2Setting { return .enabledForCompatibleDevices }
+
+    @available(iOS 15.0, tvOS 15.0, watchOS 8.0, macOS 12.0, *)
+    func testPurchaseInDevicePostsReceipt() async throws {
+        try AvailabilityChecks.iOS15APIAvailableOrSkipTest()
+
+        try await self.purchaseProductFromStoreKit()
+
+        let info = try await Purchases.shared.restorePurchases()
+        try await self.verifyEntitlementWentThrough(info)
+    }
+
+}
+
+class StoreKit1ObserverModeIntegrationTests: BaseStoreKitObserverModeIntegrationTests {
+
+    func testPurchaseOutsideTheAppPostsReceipt() async throws {
+        try self.testSession.buyProduct(productIdentifier: Self.monthlyNoIntroProductID)
+
+        let info = try await Purchases.shared.restorePurchases()
+        try await self.verifyEntitlementWentThrough(info)
+    }
+
+}
+
+@available(iOS 15.0, tvOS 15.0, watchOS 8.0, macOS 12.0, *)
+class StoreKit2ObserverModeWithExistingPurchasesTests: StoreKit1ObserverModeWithExistingPurchasesTests {
```

Comment:
Hmm we don't have any tests here, I guess the plan is to add some later? If so, should we add a comment or something? We can also delete it for now.